core/src/main/java/io/kestra/core/runners/VariableRendererFactory.java

```diff
@@ -0,0 +1,34 @@
+package io.kestra.core.runners;
+
+import io.kestra.core.runners.pebble.functions.SecretFunction;
+import io.micronaut.context.ApplicationContext;
+import io.micronaut.core.annotation.Nullable;
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+
+import java.util.List;
+
+/**
+ * Factory service for constructing {@link VariableRenderer} instances.
+ */
+@Singleton
+public class VariableRendererFactory {
+    
+    @Inject
+    private ApplicationContext applicationContext;
+    
+    @Inject
+    @Nullable
+    private VariableRenderer.VariableConfiguration variableConfiguration;
+    
+    @Inject
+    protected VariableRenderer variableRenderer;
+    
+    public VariableRenderer getDefault() {
+        return variableRenderer;
+    }
+    
+    public VariableRenderer createWithMaskedSecrets() {
+        return new VariableRenderer(applicationContext, variableConfiguration, List.of(SecretFunction.NAME));
```

Comment:
This would prevent overriding the default variable renderer which is used by one of our customer.
See https://github.com/kestra-io/kestra/pull/11559 which work around the issue, the same factory should be used here.

Reply:
Thanks, good catch! I'll wait for the PR to be merged to rebase this one!